deep_qa/training/text_trainer.py

```diff
@@ -647,31 +651,47 @@ def __get_new_embedding(self, name: str, vocab_name: str='words'):
         """
         embedding_params = self.embedding_params.pop(name)
         with tensorflow.device("/cpu:0"):
-            pretrained_file = embedding_params.pop('pretrained_embeddings_file', None)
+            pretrained_file = embedding_params.pop('pretrained_file', None)
+            projection_layer = None
             if pretrained_file:
                 embedding_layer = PretrainedEmbeddings.get_embedding_layer(
                         pretrained_file,
                         self.data_indexer,
                         embedding_params.pop('fine_tune', False),
                         name=name + '_embedding')
+
+                if embedding_params.pop('project', False):
+                    projection_layer = Dense(units=embedding_params.pop('dimension'), name=name + "_projection")
```

Comment:
Maybe add a comment for why there's no `TimeDistributed` here - we'll handle it later, so that we can more easily reuse the embeddings on different input lengths.